### PR TITLE
Remove //base dependency from sky_snapshot

### PR DESCRIPTION
**Pull request by abarth**
@jamesr @cbracken

### CODE REVIEW

File: sky/tools/sky_snapshot/main.cc [hunk 1]
```diff
@@ -1,108 +1,240 @@
...
-  std::cerr << R"USAGE(usage: sky_snapshot --packages=PACKAGES --snapshot=SNAPSHOT_BLOB <lib/main.dart>
-       [ --depfile=DEPS_FILE ] [ --build-output=BUILD_OUTPUT ] [ --help ]
-
- * PACKAGES is the '.packages' file that defines where to find Dart packages.
- * SNAPSHOT_BLOB is the file to write the snapshot into.
- * DEPS_FILE is an optional '.d' file to depedendency information into.
- * BUILD_OUTPUT optionally overrides the target name used in the DEPS_FILE.
-   (The default is SNAPSHOT_BLOB.)
-)USAGE";
+  std::cerr
```
Comment (jamesr) [inline sky/tools/sky_snapshot/main.cc]: pretty sure this is not the right technology for this sort of thing.

File: sky/tools/sky_snapshot/main.cc [hunk 2]
```diff
@@ -1,108 +1,240 @@
...
+    Dart_ExitScope();
+    Dart_ExitIsolate();
+  }
+};
+
+void InitDartVM() {
+  FTL_CHECK(Dart_SetVMFlags(arraysize(kDartArgs), kDartArgs));
+  char* error = Dart_Initialize(
+      kDartVmIsolateSnapshotBuffer, nullptr, nullptr, nullptr, nullptr, nullptr,
+      nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr);
```
Comment (jamesr) [inline sky/tools/sky_snapshot/main.cc]: i am a dart vm user, let me sing you the song of my people

null null null null null null null null null null null null

**Comment by jamesr** [comment]
lgtm

File: sky/tools/sky_snapshot/main.cc (hunk 1 above)
Comment (abarth) [inline sky/tools/sky_snapshot/main.cc]: What should I use instead?

File: sky/tools/sky_snapshot/main.cc (hunk 2 above)
Comment (abarth) [inline sky/tools/sky_snapshot/main.cc]: Yeah, it's super nutty.  IMHO, they should have a struct that does this.

Comment (abarth) [inline sky/tools/sky_snapshot/main.cc]: It's worse when you actually pass some values!

**Comment by cbracken** [comment]
lgtm

**Comment by abarth** [comment]
I sent https://codereview.chromium.org/2223463003 to make Dart_Initialize us a struct.

**Pull request merged**